### ISSUE TITLE
pusher.connection is not mocked meaning mocking a bind on connection 'state_change' is unavailable

### ISSUE THREAD

**Issue by DeadEnglish** [nikolalsvk/pusher-js-mock#28]
Currently `connection` isn't mocked within `PusherMock` so when using the following code in my component, my tests break:
```
pusher.connection.bind(
	'state_change',
	(states) => {
		console.log({states});
	}
);
```

**Comment by DeadEnglish** [comment]
As a workaround for this, the functionality that I needed to test was the `bind` and `emit` on the connection to make sure functions on connection state changes It's not the best solution but I mocked it as a `PusherChannelMock` within the code, you can see alterations on my fork (https://github.com/DeadEnglish/pusher-js-mock).

Obviously this should be mocked correctly when possible but for now it'll do until I get time to properly comeback and possibly mock the entire ConnectionManager correction.

**Comment by nikolalsvk** [comment]
Got it, @DeadEnglish, thanks for opening the issue and explaining how to work around this. If you have time, please submit a PR 🙏

**Comment by DeadEnglish** [comment]
@nikolalsvk Have created a PR for you, cheers for the switch reply 😄 #31

**Comment by jonchay** [comment]
We have an application test that requires this too! Thank you @DeadEnglish for pushing this through!

**Comment by nikolalsvk** [comment]
Released the new changes by @DeadEnglish in [0.3.3 version](https://github.com/nikolalsvk/pusher-js-mock/releases/tag/0.3.3), enjoy!

**Comment by jonchay** [comment]
This seems to be working for TS but not for JS counterpart. 
https://github.com/nikolalsvk/pusher-js-mock/issues/28

I had to redefine the mock to get this to work:
```
pusherMock.connection = new PusherChannelMock('connection');
pusherMock.connection.emit(...)
```
